hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactUtil.java

```diff
@@ -25,24 +25,32 @@
  * Utility methods to be used in compact serialization
  */
 public final class CompactUtil {
+    // To determine classes that we won't serialize with zero-config serialization
+    private static final UnsupportedPackagePrefix[] UNSUPPORTED_PACKAGE_PREFIXES = new UnsupportedPackagePrefix[]{
+            new UnsupportedPackagePrefix("java"),
```

Comment:
java.lang too? ;)

Reply:
Yeah, that was the decision :/
We still continue to support the classes we support under all those packages (java.lang.String etc.), but we decided to be extra restrictive for the others

Reply:
As discussed on Slack I would like to see a best effort ser/de for List, Set and Map, similar to support of `Character` or enums. 

I don't mean to add them to CompactReader/Writer API.

Reply:
OK, I will add that support but we will be able to do that even if we merge this PR. 

We hit this code when we are about to serialize a class as compact. So, if the type of the class is one of the ones that we support, this code is not run. For those interfaces/classes, we will add support to the reflective compact serializer, and add the appropriate readers/writers without checking the package name here

Reply:
@frant-hartm can you please merge this PR? As I said, we will support these three types, but merging this PR won't affect them. 

I am currently working on supporting List, Set, and Map in reflective and record serializers.